src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -172,10 +202,45 @@ static void RenameExpression(ParsedExpression &expr, RenameColumnInfo &info) {
 	    expr, [&](const ParsedExpression &child) { RenameExpression((ParsedExpression &)child, info); });
 }
 
+unique_ptr<CatalogEntry> TableCatalogEntry::RenameGeneratedColumn(ClientContext &context, RenameColumnInfo &info,
+                                                                  const TableColumnInfo &rename_info) {
+	D_ASSERT(rename_info.column_type == TableColumnType::GENERATED);
```

Comment:
I thought they were different enough, but I'm happy to change it if you feel they are too similar
When we change a generated column's name, there aren't any additional checks (other than checking if the new name doesn't collide with existing names of generated columns - which I just remembered)

Whereas for a regular column we need to rename the columnrefs in the generated columns' expressions that reference this column, and check constraints.
Because of all the checks in between the copying over of columns+constraints I felt returning early wasn't really an option

Reply:
Can generated columns not reference other generated columns?

Reply:
I think I'll focus on fixing all the other issues first, then I'll start working on allowing generated columns to be referenced within generated expressions